### DISCUSSION TITLE
Debugging / workflow tools

### DISCUSSION THREAD

**Discussion by leotrs** [ManimCommunity/manim#2774] [Suggestions and Proposals]
## Description of proposed feature
The process of building animations with manim is quite arduous. One thing that I have identified that would help me a lot is to have the option to show guidelines on screen. These are optional and easily toggle-able thin lines rendered on top of the scene to serve as position and coordinate guides. Basically any other software for design/slideshows/etc allows the use of rulers and guidelines.

Other similar tools would be:
1. A small overlay in a corner showing the animation number. Sometimes, after rendering a video I have found that I have to shrink or expand the self.wait() time in the middle of two animations. But finding the correct self.wait() call among 100s of lines of code usually takes longer than I'd care for. Showing the animation number in real time on the video would make this easier.

2. Similar to above, showing an overlay with the frame number, and the current in-video time (and optionally the time left until the end of the video).

## How can the new feature be used?
I see two ways of using it: through config and through a CLI flag.
```python
from manim import *

# All of these would have sensible defaults so you
# don't have to set them all each time. You could
# also just set them once in your user .cfg file.
config.guidelines_x = np.arange(-14, 14)    # Show thin vertical lines on top of all other mobjects
config.guidelines_y = np.arange(-8, 9)      # Show thin horizontal lines on top of all other mobjects
config.frame_number_overlay = True          # Show an overlay with the current frame number
config.debug_mode = True                    # If False, all of the above will be ignored.

class MyScene(Scene):
    def construct(self):
        # same code as usual, nothing changes
```
And via CLI flag, one could easily toggle debug mode tools on and off:
```bash
manim -pql scene.py MyScene --debug   # equivalent to writing config.debug_mode = True
manim -pql scene.py MyScene           # by default all debug options are off or False
```

## Additional comments
I'm trying to gauge interest from other users to see if these are worth implementing.

I think that perhaps most of these will be made obsolete by some medium-term or long-term changes to manim. But as of right now, the workflow can be pretty painful and these are fairly easy to implement. So these are meant to help users _right now_, and not to delay the implementation of better workflows later on.

**Comment by huguesdevimeux** [comment]
This seems to me a very good idea. I'm just reserved on the idea of adding this directly to the scene as an object.

I think it would make more sense at any level (including the practical one) to put this in an totally external way, by superposing the normal frame with a debug frame, I don't know.

**Comment by leotrs** [comment]
Ok I see your point. The way I'm proposing would add a lot of Mobjects to the scene that are there only for debug purposes. And this may be an obstruction for the end user.

**Comment by huguesdevimeux** [comment]
I started to work on something like this, an overtay at the right top bottom in the videos that will display debug infos.
